### PR TITLE
ASMs3Dmx: add support for a separate geometry basis

### PR DESCRIPTION
**Pull request by akva2**
Same as https://github.com/OPM/IFEM/pull/623 but in 3D.

They have a common-ish commit (the forceItg param) so they are sort of dependent.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams ifem-stokes=240 please

**Comment by kmokstad** [comment]
yes, rebase it on top of #623 (after merging that one), and I'll review it, so I don't have to check the same commits twice ;)

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Comment by akva2** [comment]
jenkins build this with downstreams please

File: src/ASM/ASMs3D.C [hunk 1]
```diff
@@ -1952,12 +1952,16 @@ bool ASMs3D::getParameterDomain (Real2DMat& u, IntVec* corners) const
 
 
 const Vector& ASMs3D::getGaussPointParameters (Matrix& uGP, int dir, int nGauss,
-					       const double* xi) const
+                                               const double* xi,
+                                              const Go::SplineVolume* spline) const
```
Comment (kmokstad) [inline src/ASM/ASMs3D.C:1956]: check indent

**Review by kmokstad — APPROVED** [review]

File: src/ASM/ASMs3D.C (hunk 1 above)
Comment (akva2) [reply src/ASM/ASMs3D.C]: fixed.

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Pull request merged**